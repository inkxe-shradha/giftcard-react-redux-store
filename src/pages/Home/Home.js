import React from "react";
import "./home.scss";
import CustomCarousel from "../../components/Carousel/Customcarousel";
import { Divider, Chip } from "@mui/material";
import StaticCoupons from "../../components/StaticCoupons/StaticCoupons";

const Home = () => {
  return (
    <div>
      <CustomCarousel />
      <Divider sx={{ marginTop: 5 }}>
        <Chip label="Most Viewed" />
      </Divider>
      <StaticCoupons />
    </div>
  );
};

Home.propTypes = {};

export default Home;
