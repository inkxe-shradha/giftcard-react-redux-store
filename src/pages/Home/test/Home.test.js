import { createTheme, ThemeProvider } from "@mui/material";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Home from "../Home";

describe("Home component suite case", () => {
  const theme = createTheme();
  const customRender = (props = {}) => {
    return mount(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Home {...props} />
        </BrowserRouter>
      </ThemeProvider>
    );
  };
  it("Should render the component snapshot", () => {
    const view = customRender();
    expect(view.html()).toMatchSnapshot();
  });
});
