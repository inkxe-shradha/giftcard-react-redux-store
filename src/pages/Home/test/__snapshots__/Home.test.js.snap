// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component suite case Should render the component snapshot 1`] = `"<div><div class=\\"container-fluid\\"><div class=\\"carousel-root\\"><div class=\\"carousel carousel-slider\\" style=\\"width: 100%;\\"><ul class=\\"control-dots\\"><li class=\\"dot selected\\" value=\\"0\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"slide item 1\\"></li><li class=\\"dot\\" value=\\"1\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"slide item 2\\"></li><li class=\\"dot\\" value=\\"2\\" role=\\"button\\" tabindex=\\"0\\" aria-label=\\"slide item 3\\"></li></ul><button type=\\"button\\" aria-label=\\"previous slide / item\\" class=\\"control-arrow control-prev control-disabled\\"></button><div class=\\"slider-wrapper axis-horizontal\\"><ul class=\\"slider animated\\" style=\\"transform: translate3d(0,0,0); transition-duration: 350ms;\\"><li class=\\"slide selected previous\\"><div><img class=\\"carousel-img\\" src=\\"https://source.unsplash.com/featured/900x450?amazon\\" alt=\\"amazon\\"><button class=\\"legend\\"> Grab it</button></div></li><li class=\\"slide\\"><div><img class=\\"carousel-img\\" src=\\"https://source.unsplash.com/featured/900x450?coupon\\" alt=\\"flipkart\\"><button class=\\"legend\\"> Grab it</button></div></li><li class=\\"slide\\"><div><img class=\\"carousel-img\\" src=\\"https://source.unsplash.com/featured/900x450?google\\" alt=\\"google\\"><button class=\\"legend\\"> Grab it</button></div></li></ul></div><button type=\\"button\\" aria-label=\\"next slide / item\\" class=\\"control-arrow control-next\\"></button><p class=\\"carousel-status\\">1 of 3</p></div></div></div><div class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-19bfbp2-MuiDivider-root\\" role=\\"separator\\"><span class=\\"MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper\\"><div class=\\"MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gac2fo-MuiChip-root\\"><span class=\\"MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label\\">Most Viewed</span></div></span></div><div class=\\"MuiContainer-root MuiContainer-maxWidthLg makeStyles-container-2 css-1oqqzyl-MuiContainer-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root\\"><h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root\\">Most Viewed Coupons</h4></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-10imocd-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root\\"><img class=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root\\" src=\\"https://pngimg.com/uploads/amazon/amazon_PNG21.png\\" height=\\"140\\" alt=\\"Amazon Card\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><div class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root\\">Amazon Gift (5000)</div><p class=\\"MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root\\">Amazon Gift Cards are the Perfect Gift, Every Time. Use the eBay Gift Card to shop from millions of items .....</p></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root\\"><img class=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root\\" src=\\"https://i.pinimg.com/originals/15/aa/16/15aa1678d4ee5615c5c53ed5c9968126.png\\" height=\\"140\\" alt=\\"FlipKart\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><div class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root\\">Flip-kart (5000)</div><p class=\\"MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root\\">Flipkart Pvt Ltd. is an e-commerce company based in Bengaluru, India. Founded by Sachin Bansal and Binny Bansal in 2007 ...</p></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root\\"><img class=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root\\" src=\\"https://img.etimg.com/thumb/msid-52350287,width-300,imgsize-34171,resizemode-4/myntra-app-faces-tech-glitch-spams-users-with-notifications.jpg\\" height=\\"140\\" alt=\\"Myntra\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><div class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root\\">Myntra</div><p class=\\"MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root\\">Myntra Gift Cards are the Perfect Gift, Every Time. Use the Myntra Gift Card to shop from millions of items in Electronics...</p></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root\\"><img class=\\"MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root\\" src=\\"https://cms.qz.com/wp-content/uploads/2015/05/zomatos-4th-logo-2012-14.png?w=900&amp;h=900&amp;crop=1&amp;strip=all&amp;quality=75\\" height=\\"140\\" alt=\\"Zomato\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><div class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root\\">Zomato (5000)</div><p class=\\"MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root\\">Zomato. Zomato is an Indian restaurant search and discovery service founded in 2008 by Deepinder Goyal ....</p></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root\\" align=\\"center\\"><a class=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/gift-cards\\">View More<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a></div></div></div></div></div></div>"`;
